src/Tools/dotnet-monitor/Egress/Configuration/EgressProviderValidateOptions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration
+{
+    /// <summary>
+    /// Validates the settings within a <typeparamref name="TOptions"/> instance
+    /// using data annotation validation.
+    /// </summary>
+    internal sealed class EgressProviderValidateOptions<TOptions> :
+        IValidateOptions<TOptions> where TOptions : class
+    {
+        public ValidateOptionsResult Validate(string name, TOptions options)
+        {
+            ValidationContext validationContext = new ValidationContext(options);
+            ICollection<ValidationResult> results = new Collection<ValidationResult>();
+            if (!Validator.TryValidateObject(options, validationContext, results, validateAllProperties: true))
+            {
+                IList<string> failures = new List<string>();
+                foreach (ValidationResult result in results)
+                {
+                    if (ValidationResult.Success != result)
+                    {
+                        failures.Add(result.ErrorMessage);
+                    }
+                }
+
+                if (failures.Count > 0)
```

Comment:
Not sure that this is correct/incorrect behavior, but I thought it was interesting that we have the check for failures.Count > 0 in order to return ValidateOptionsResult.Fail(failures). In theory, since we're in the current conditional (stating that TryValidateObject returned false), we already know we failed. In which case, is the check redundant? And if not, and failures.Count == 0, is the correct behavior actually to return ValidateOptionsResult.Success? Given that we failed to validate the object, that to me would seem odd. Something just felt off when I'm reading this, so I figured I'd make a note of it.

Reply:
Because the validators _could_ return `ValidationResult.Success` entries, we should protect against that. And if all of the entries are success (or an empty list), then we can presume it is valid.